syntax = "proto3";

package storage;

message Key { string key = 1; }

service StorageService {
  rpc SaveValue (Value) returns (Empty) {}
  rpc GetValue (Key) returns (Value) {}
  rpc GetAllValues (Empty) returns (ValueList) {}
}

message Empty {}

message ValueList { repeated Value values = 1; }


message Value {
  string key = 1;
  string data = 2;
}